### ISSUE TITLE
Instanced .blend/GLTF scenes lose all children after update until .tscn is reopened

### ISSUE THREAD

**Issue by Yahkub-R** [godotengine/godot#94056]
### Tested versions

- Reproducible in: v4.3.beta2.official [b75f048]
- Reproducible in: v4.3.beta.custom_build [e57312d]
- Not reproducible in: v4.3.beta.custom_build [1e6b11d]
- Not reproducible in: v4.2.dev.custom_build [524e7ac]
- Not reproducible in: v4.2.2.stable.official [15073af]

### System information

Godot v4.2.2.stable - Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4090 (NVIDIA; 32.0.15.5599) - Intel(R) Core(TM) i9-14900K (32 Threads)

### Issue description

When making modifications to a .blend file, I noticed that upon reimport any child nodes that were previously there disappear until the scene is reloaded.   From a cursory glance I believe the issue has to do with [editor_node.cpp:6025 ](https://github.com/godotengine/godot/blob/42e5b3ac2da07d2105c775977b39e6949c723ded/editor/editor_node.cpp#L6025) where the addition_list is empty

### Steps to reproduce

Create new project
Instantiate an imported .blend scene
add a child node to it
save scene
modify the .blend file (anything to trigger re-import)
observe missing child node
close and re-open the scene
observe missing child node reappears.

### Minimal reproduction project (MRP)

[MRP.zip](https://github.com/user-attachments/files/16121065/MRP.zip)

**Comment by Yahkub-R** [comment]
Removing the following condition fixes the issue, however the re import only works once, after the first import any subsequent do not update the scene. So Something else up with hot-reloading.
https://github.com/godotengine/godot/blob/e57312d84eecb7672973a36c0938ed52a3b15474/editor/editor_node.cpp#L4345

**Comment by akien-mga** [comment]
CC @SaracenOne @Hilderin

**Comment by Hilderin** [comment]
I tested this PR and it seems to work nicely.

I just found a little side-effect and I think that way the set for ownership was there. If the child added is another scene with child, I get the error: `Invalid owner. Owner must be an ancestor in the tree.` when reimporting the .blend file.

![image](https://github.com/godotengine/godot/assets/81109165/66686e0f-0528-48bd-86d2-5b5cbaec3dc3)

![image](https://github.com/godotengine/godot/assets/81109165/e4c44802-c9bf-47d7-9bb4-ec5a2e595d21)


There is a simple modified version of the MRP:
[test-godot-blender-reimport-missing-node.zip](https://github.com/user-attachments/files/16144191/test-godot-blender-reimport-missing-node.zip)
